Xamarin.Forms.Core/CornerRadius.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System.Diagnostics;
+
+namespace Xamarin.Forms
+{
+	[DebuggerDisplay ("TopLeft={TopLeft}, TopRight={TopRight}, BottomLeft={BottomLeft}, BottomRight={BottomRight}")]
+	[TypeConverter (typeof (CornerRadiusTypeConverter))]
+	public struct CornerRadius
+	{
+		public double TopLeft { get; private set; }
+		public double TopRight { get; private set; }
+		public double BottomLeft { get; private set; }
+		public double BottomRight { get; private set; }
+
+		internal bool IsDefault
+		{
+			get { return TopLeft == 0 && TopRight == 0 && BottomLeft == 0 && BottomRight == 0; }
```

Comment:
`(0,0,0,0)` isn't a good default for `CornerRadius`. Some controls (android buttons, e.g.) are rounded by default. if you want to make them with square angles, you will set `button.CornerRadius = new CornerRadius (0,0,0,0)`. the renderer will then interpret this as `Default` and will use the default (which is rounded) for the corners.

as this is a `struct`, you can not have default values for fields (they have to be `0`), so you have to use a flag (`false` by default) that you set to `true` the non-default ctors. `default(CornerRadius)` will return a struct with that field unchanged...

if it's unclear, I can do that for you